Kernel: auth.erl: skip cookie file permission checks

When the Erlang runtime is run in Nanos, there are no real owners of files
in the filesystem, thus there is no need to check file permissions for the
cookie. This patch fixes the "Cookie file /.erlang.cookie must be
accessible by owner only" error.

diff -upr otp_orig/lib/kernel/src/auth.erl otp/lib/kernel/src/auth.erl
--- otp_orig/lib/kernel/src/auth.erl	2020-05-13 10:47:33.000000000 +0200
+++ otp/lib/kernel/src/auth.erl	2020-05-26 16:20:59.158885024 +0200
@@ -345,8 +345,6 @@ make_error(Name, Reason) ->
 
 check_attributes(Name, Type, _Mode, _Os) when Type =/= regular ->
     {error, "Cookie file " ++ Name ++ " is of type " ++ Type};
-check_attributes(Name, _Type, Mode, {unix, _}) when (Mode band 8#077) =/= 0 ->
-    {error, "Cookie file " ++ Name ++ " must be accessible by owner only"};
 check_attributes(_Name, _Type, _Mode, _Os) ->
     ok.
